### DISCUSSION TITLE
Fire TV Stick: MXPlayer doesn't work well; recommend VLC PLayer (with an audio sync fix)

### DISCUSSION THREAD

**Discussion by DVoorhees1957** [GerryDazoo/Slinger#487] [General]
After some additional research and very useful discussion with @GerryDazoo, I've determined that MXPlayer simply has unresolvable problems when trying to use it as the media player on a Fire TV Stick. For details, see discussion #259. So I went back to trying to use VLC Player, which I had initially abandoned because of a very painful audio lag relative to the video on almost everything I watched.

Luckily, there's a fix for the latter. Here is the information @GerryDazoo provided about how to access the "audio delay" setting in VLC (I think you have to be connected to Slinger and watching a video stream first):

```
Click in the player window; this will display the pause button and some other controls
Click on the icon in the lower left hand corner (who knows what it stands for)
Click on the Audio drop-down
Click on Audio Delay
```
In my specific case, I had to set the audio delay to "-350ms" (minus 350 ms); that gets the lip movement and audio pretty much in sync, at least for the show I was testing with. I'll have to see if that delay needs to be adjusted for each program separately - hopefully not. There's a way to save the setting to be used for all shows, not just the current one.

**Comment by socelillyhammer** [comment]
its like im watching an asian movie dubbed in english.  Thank you for this. looking forward to testing this out. the delay is painful.

**Comment by tom76017** [comment]
My Firestick 4K works great with MXPlayer.  No audio delays.  VLC is terrible with huge audio delays.  Strange that we get differing results.  And, the original Slingplayer also works well, although I prefer Slinger and my Network IP Remote connected directly to my TiVo.

**Reply by DVoorhees1957** [reply]
The MXPlayer issue isn't audio lag; it's actually great at that (and VLC Player is bad). It's that MXPlayer sends multiple connection commands to Slinger, even after its initial request is successful, so it basically uses up all of the maxstreams that you might have set (if you have maxstreams set to a number greater than 1). So the problem comes that you can't connect with any other devices to the same Slingbox - MXPlayer basically breaks the capability to have the same Slingbox serve multiple devices...

**Reply by tom76017** [reply]
Not really seeing that issue either.  It is a local connection.

VVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVV
...
Slinginfo  "350/500/M1/M2" 16 30 192.168.11.70 5301 3072 10
Slingbox-M1 Starting Stream for  192.168.11.1:51778
Connecting... ('192.168.11.70', 5301) Control
Slingbox-M1 Selecting Video Source 1
Keycodes= [4, 5, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 21, 24, 26, 27, 28, 29, 30, 31, 32, 35, 36, 37, 38, 39, 40, 41, 42, 46, 50, 72, 73, 74, 75, 76, 77, 78, 79]
VideoParameters: Resolution= 16 FrameRate= 30 VideoBandwidth= 1984 VideoSmoothness= 59 IframeRate= 4 AudioBitRate= 64
Connecting... ('192.168.11.70', 5301) Stream
Slingbox-M1 Stream started at 01/09/2023, 08:59:23.348  744 0
.........01/09/2023, 09:00:56.215  Slingbox-M1 1 Clients.192.168.11.1:51778 
.........01/09/2023, 09:02:28.652  Slingbox-M1 1 Clients.192.168.11.1:51778 
.........01/09/2023, 09:04:00.980  Slingbox-M1 1 Clients.192.168.11.1:51778 
.........01/09/2023, 09:05:33.027  Slingbox-M1 1 Clients.192.168.11.1:51778 
.........01/09/2023, 09:07:04.835  Slingbox-M1 1 Clients.192.168.11.1:51778 
.........01/09/2023, 09:08:38.372  Slingbox-M1 1 Clients.192.168.11.1:51778 
.........01/09/2023, 09:10:11.609  Slingbox-M1 1 Clients.192.168.11.1:51778 
.........01/09/2023, 09:11:44.373  Slingbox-M1 1 Clients.192.168.11.1:51778 
.........01/09/2023, 09:13:17.394  Slingbox-M1 1 Clients.192.168.11.1:51778 
.........01/09/2023, 09:14:49.407  Slingbox-M1 1 Clients.192.168.11.1:51778 
.........01/09/2023, 09:16:21.559  Slingbox-M1 1 Clients.192.168.11.1:51778 
.........01/09/2023, 09:17:54.366  Slingbox-M1 1 Clients.192.168.11.1:51778 
.........01/09/2023, 09:19:27.310  Slingbox-M1 1 Clients.192.168.11.1:51778 
01/09/2023, 09:19:27.680  Slingbox-M1 Stream Terminated for  192.168.11.1:51778
Slingbox-M1 Shutting down connections
Slingbox-M1 Logging Out
Streamer:  Slingbox-M1 Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
01/09/2023, 09:20:04.843   Streaming request from ('192.168.11.1', 52482)
...
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

**Reply by GerryDazoo** [reply]
What version of MXPlayer are you running? I've seen the multi connect issue on two different firesticks.

**Reply by tom76017** [reply]
Firestick 4K - Ver 1.36.11 (ARMv7 NEON)

**Reply by GerryDazoo** [reply]
Same here! I wonder what the delta is?

**Reply by DVoorhees1957** [reply]
My config: Fire TV Stick 4K running Fire OS 6.2.9.4; MXPlayer v1.36.11 (doesn't show any "ARMv7 NEON" after it); Slinger running on a Windows 7 laptop on the same LAN as the Fire TV; Slingbox 500 running on a different LAN than Slinger and the Fire TV.

@tom76017 Not that it should matter, but are your Fire TV, Slinger, and the Slingbox all on the same LAN?

**Reply by tom76017** [reply]
Yes, local LAN for all.  Slinger is on a local Debian server.  I bought the server on eBay for $50 and installed Debian on a drive I salvaged via an upgrade to Win11 PC.  It's also my Samba, Apache2, and other processes.  I'll try connecting my Firestick to an iOS hotspot and see how it works externally.
![image](https://user-images.githubusercontent.com/26171888/211413193-c731759c-5d57-41b8-82a5-da1ff94e7c9c.png)

VVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVV
Edit: Follow-up 5 minutes later.  No issues with the hot spot.  No restarts, no audio sync issues.  The server was an HP Elite 8000 desktop (Intel(R) Core(TM)2 Duo CPU) converted to Linux.  Slinger puts no load on the CPU.

**Reply by tom76017** [reply]
FYI 
![image](https://user-images.githubusercontent.com/26171888/211417157-2178f62e-f320-4e36-961e-4260b3c1d66b.png)

**Reply by DVoorhees1957** [reply]
Ah, the About info in the MXPlayer app is what you were referencing - mine shows the same thing. So the only "obvious" difference between your config and mine (not sure about Gerry's) is that you're running Slinger on top of Linux, rather than Windows 7 (Python 3.8.10). Maybe there's some difference in the Python3 implementation, or underlying network protocols, between Linux and Win7...?

**Reply by socelillyhammer** [reply]
@tom76017 what? the original slingplayer? how? are you using it with the SB 500?

**Reply by GerryDazoo** [reply]
I have slinger running on windows and Linux and I see the same results.  I run the exe and Python and see the same results. So sorry, no cigar. I'm voting for a bug in MXPlayer.

**Reply by GerryDazoo** [reply]
Lilly. Slingbox Player 5.0.0.83. Is working for me. It only works for playing content on the LAN, no place shifting.

**Reply by robosmith** [reply]
I ONLY see problems with MxPlayer (Fire TV stick from Pro HD with Slinger on Pi 4) when it repeatedly drops the stream (proly due to Internet congestion over 3000mi link) and I try to reconnect. For some unknown reason, it usually will not reconnect until 2-4 tries later. I've seen as many as 8 orphaned streams.  I wonder if it's dropping the streams due to buffer overrun.

VLC usually reconnects in less than 30 seconds with ONE try. I use the -250ms audio delay setting.

Otherwise, with low net congestion, MxPlayer is SOLID and better pic than VLC or Slingplayer was.
